### DISCUSSION TITLE
Any way to extract airspaces from the navdata sql?

### DISCUSSION THREAD

**Discussion by sanjaypopcorn** [albar965/littlenavmap#1088] [Q&A] [answered]
I'm sorry if this isn't the place to ask this but im working on a personal project of mine and need a way to extract airspace data and turn them into GNG coordinates. Specifically the geometry BLOB binary data stored in the BOUNDARY table of the sql db. (I dont know if this is a roundabout way of doing this) I'd be greatful if any of you cound point me in the right direction.

**Comment by albar965** [comment, marked as answer]
The format is quite simple. Reading and writing is implemented here:

https://github.com/albar965/atools/blob/master/src/fs/common/binarygeometry.h
https://github.com/albar965/atools/blob/master/src/fs/common/binarymsageometry.cpp

It starts with an unsigned 32 bit integer which contains the number of points. Then following single precision floating point number pairs for lon/lat.

The geometry is pre-calculated which means that circles and arcs are converted to polygons.

**Reply by sanjaypopcorn** [reply]
Sir, sadly I have very limited experience in c++ and none in binary🥲 so im having trouble parsing what the function actually does. If its not too much trouble, could this be done in python?
Nonetheless your answer was greatly appreciated :)

**Reply by albar965** [reply]
Should be doable in all languages but frankly I have no clue about Python.

**Reply by sanjaypopcorn** [reply]
Do you reakon theres another way to extract the lat,long of the polygons from the db? Perhaps if you have some free time could you add the functionality in LNM or a standlaone app. I know im asking alot.
Thank you for your help still.

**Reply by albar965** [reply]
https://www.geeksforgeeks.org/reading-binary-files-in-python/

The format is really totally simple. 😉

**Reply by sanjaypopcorn** [reply]
I think i figured it out, thank you so much sir. 🫡
